### PR TITLE
Communication Page Refactor and Edit

### PR DESCRIPTION
**Pull request by sam-warren**
## Status
**Ready**

## Fixes or Implements [AB#8216](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/8216)
- [X] Enhancement
- [ ] Bug
- [ ] Documentation

## Description:
- Remove form at top of page and replaced it with a modal
- Modal serves as an add and edit form
- Improve validation logic
- Wire up add communication
- Set up editing features

### UI Changes
YES

### Browsers Tested
- [X] Chrome
- [ ] Safari
- [ ] Edge
- [ ] Firefox

Unit tests for front end not completed due to lack of Jest environment on admin web client.

New Communications Page:
table

New Communication Item:
new

Edit Existing Item:
edit

Invalid Dates:
invalid

### CODE REVIEW

File: Apps/AdminWebClient/src/ClientApp/src/services/restCommunicationService.ts [hunk 1]
```diff
@@ -52,6 +52,23 @@ export class RestCommunicationService implements ICommunicationService {
         });
     }
 
+    public update(communiucation: Communication): Promise<void> {
+        return new Promise((resolve, reject) => {
+            this.http
+                .put<RequestResult<Communication>>(
+                    `${this.BASE_URI}/`,
+                    communiucation
```
Comment (sslaws) [inline Apps/AdminWebClient/src/ClientApp/src/services/restCommunicationService.ts:60]: typo: communiucation

File: Apps/WebClient/src/ClientApp/src/models/communication.ts [hunk 2]
```diff
@@ -5,6 +5,12 @@ export default interface Communication {
     // Gets or sets the communication subject.
     subject: string;
 
-    // Gets or sets the communication text.
-    text: string;
+  // Gets or sets the communication text.
+  text: string;
```
Comment (FuriousLlama) [inline Apps/WebClient/src/ClientApp/src/models/communication.ts:9]: nit, indentation

File: Apps/WebClient/src/ClientApp/src/services/interfaces.ts [hunk 3]
```diff
@@ -122,8 +122,8 @@ export interface IUserCommentService {
 }
 
 export interface ICommunicationService {
-    initialize(http: IHttpDelegate): void;
-    getActive(): Promise<RequestResult<Communication>>;
+  initialize(http: IHttpDelegate): void;
```
Comment (FuriousLlama) [inline Apps/WebClient/src/ClientApp/src/services/interfaces.ts:125]: indentation

Comment (sam-warren) [reply Apps/WebClient/src/ClientApp/src/services/interfaces.ts]: fixed

File: Apps/WebClient/src/ClientApp/src/models/communication.ts (hunk 2 above)
Comment (sam-warren) [reply Apps/WebClient/src/ClientApp/src/models/communication.ts]: fixed

File: Apps/AdminWebClient/src/ClientApp/src/services/restCommunicationService.ts (hunk 1 above)
Comment (sam-warren) [reply Apps/AdminWebClient/src/ClientApp/src/services/restCommunicationService.ts]: fixed

**Pull request merged**